docs/book/src/cronjob-tutorial/testdata/project/controllers/suite_test.go

```diff
@@ -0,0 +1,154 @@
+(this example runs a simple envtest cluster of just etcd and the apiserver by default).
+
+After the cluster starts running, we’ll need a client to connect to it.
+The autogenerated suite_test.go file will do this for you and also add your custom Kind to the scheme the client will use.
+This way, the client will be able to perform CRUD operations on instances of your custom Kind.
+*/
+
+var cfg *rest.Config
+var k8sClient client.Client // You'll be using this client in your tests.
+var testEnv *envtest.Environment
+
+var _ = BeforeSuite(func(done Done) {
+	logf.SetLogger(zap.LoggerTo(GinkgoWriter, true))
+
+	// Step 1: Configure envtest.
+	// Include CRDDirectoryPaths arg if you're testing a new CRD.
+	By("bootstrapping test environment")
+	testEnv = &envtest.Environment{
+		CRDDirectoryPaths: []string{filepath.Join("..", "config", "crd", "bases")},
+	}
+
+	// Step 2: Start envtest!
+	var err error
+	cfg, err = testEnv.Start()
+	Expect(err).ToNot(HaveOccurred())
+	Expect(cfg).ToNot(BeNil())
+
+	// Step 3: If using new CRD, add its corresponding Kind to the client-go
+	// k8s scheme. After this, it will be available in your k8s clientset.
+	err = batchv1.AddToScheme(scheme.Scheme)
+	Expect(err).NotTo(HaveOccurred())
+
+	// Step 4: Get that k8s client and use it for tests!
+	k8sClient, err = client.New(cfg, client.Options{Scheme: scheme.Scheme})
+	Expect(err).ToNot(HaveOccurred())
+	Expect(k8sClient).ToNot(BeNil())
+
+	/*
+		One thing that this autogenerated file is missing, however, is a way to actually start your controller.
+		The code above will set up a client for interacting with your custom Kind,
+		but will not be able to test your controller behavior.
+		If you want to test your custom controller logic, you’ll need to add some familiar-looking manager logic
+		to your BeforeSuite() function, so you can register your custom controller to run on this test cluster.
+
+		You may notice that the code below runs your controller with nearly identical logic to your CronJob project’s main.go!
+		The only difference is that the manager is started in a separate goroutine so it does not block the cleanup of envtest
+		when you’re done running your tests.
+
+		Once you've added the code below, you can actually delete all the code under step 4 above, because you can get k8sClient from the manager
+		defined in step 5.
+	*/
+
+	// Step 5: Run your test controller
+	// +kubebuilder:scaffold:scheme
+	mgr, err := ctrl.NewManager(cfg, ctrl.Options{
+		Scheme: scheme.Scheme,
+	})
+	Expect(err).ToNot(HaveOccurred())
+	err = (&CronJobReconciler{
+		Client: mgr.GetClient(),
+		Log:    ctrl.Log.WithName("controllers").WithName("CronJob"),
+	}).SetupWithManager(mgr)
+
+	Expect(err).ToNot(HaveOccurred())
+	go func() {
+		err = mgr.Start(ctrl.SetupSignalHandler())
+		Expect(err).ToNot(HaveOccurred())
+	}()
+
+	k8sClient = mgr.GetClient()
+	Expect(k8sClient).ToNot(BeNil())
```

Comment:
Here is the before implementation for each test that should be executed. 
So, we need to add a test for each controller and then, the tests/checks in the controller.